### PR TITLE
Simplify conditional.py

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY
Quick perf test:
```python
In [1]: import timeit
In [2]: from ansible.playbook.conditional import Conditional
In [3]: from ansible.template import Templar
In [4]: from ansible.parsing.dataloader import DataLoader
In [5]: t = Templar(None)
In [6]: c = Conditional(loader=DataLoader())
In [7]: c.when = ["x == y"]*100
In [8]: %timeit c.evaluate_conditional(templar=t, all_vars={"x": 1, "y": 1})
```
Before:
```
89.2 ms ± 1.5 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

After:
```
58.9 ms ± 820 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

ci_complete

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`lib/ansible/playbook/conditional.py`

### CODE REVIEW

File: lib/ansible/playbook/conditional.py [hunk 1]
```diff
@@ -107,83 +102,22 @@ def evaluate_conditional_with_result(self, templar: Templar, all_vars: dict[str,
...
         return result, false_condition
 
-    def _check_conditional(self, conditional, templar, all_vars):
-        '''
-        This method does the low-level evaluation of each conditional
-        set on this object, using jinja2 to wrap the conditionals for
-        evaluation.
-        '''
-
+    def _check_conditional(self, conditional: str, templar: Templar, all_vars: dict[str, t.Any]) -> bool:
```
Comment (bcoca) [inline lib/ansible/playbook/conditional.py:88]: iirc, conditional can be str or bool

File: lib/ansible/playbook/conditional.py [hunk 2]
```diff
@@ -107,83 +102,22 @@ def evaluate_conditional_with_result(self, templar: Templar, all_vars: dict[str,
...
         templar.available_variables = all_vars
-
         try:
-            # if the conditional is "unsafe", disable lookups
-            disable_lookups = hasattr(conditional, '__UNSAFE__')
-            conditional = templar.template(conditional, disable_lookups=disable_lookups)
-
-            if not isinstance(conditional, text_type) or conditional == "":
+            conditional = templar.template(conditional)
+            if not isinstance(conditional, str) or conditional == "":
```
Comment (bcoca) [inline lib/ansible/playbook/conditional.py]: this is why this return 'works' here as if conditional is not a string it has to be a bool, this is probably why the existing code got into soo many issues.

File: lib/ansible/playbook/conditional.py [hunk 3]
```diff
@@ -107,83 +102,22 @@ def evaluate_conditional_with_result(self, templar: Templar, all_vars: dict[str,
...
-        # make sure the templar is using the variables specified with this method
         templar.available_variables = all_vars
-
         try:
-            # if the conditional is "unsafe", disable lookups
-            disable_lookups = hasattr(conditional, '__UNSAFE__')
-            conditional = templar.template(conditional, disable_lookups=disable_lookups)
-
-            if not isinstance(conditional, text_type) or conditional == "":
+            conditional = templar.template(conditional)
```
Comment (bcoca) [inline lib/ansible/playbook/conditional.py]: no need to template the conditional here, we are doing it below after interpolation, this changes teh result as then we x2 template the conditonal and it's result

File: lib/ansible/playbook/conditional.py [hunk 4]
```diff
@@ -67,123 +61,50 @@ def evaluate_conditional(self, templar: Templar, all_vars: dict[str, t.Any]) ->
...
-            disable_lookups = hasattr(conditional, '__UNSAFE__')
-            conditional = templar.template(conditional, disable_lookups=disable_lookups)
+            if templar.is_template(conditional):
+                display.warning(
+                    "conditional statements should not include jinja2 "
+                    "templating delimiters such as {{ }} or {%% %%}. "
+                    "Found: %s" % conditional
+                )
+                conditional = templar.template(conditional)
+                if isinstance(conditional, bool):
```
Comment (bcoca) [inline lib/ansible/playbook/conditional.py:99]: we could probably 'outdent' this if/else as it would also apply to untemplated conditional

Comment (mkrizek) [reply lib/ansible/playbook/conditional.py:99]: That wouldn't get here, it is already handled above in `evaluate_conditional_with_result`.

Comment (bcoca) [reply lib/ansible/playbook/conditional.py:99]: tempted to move both into just the post_validate_X_when: functions

**Review by bcoca — APPROVED** [review]

**Review by sivel — APPROVED** [review]

**Pull request merged**